From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Branson <andrew.branson@jolla.com>
Date: Wed, 5 Aug 2026 17:49:09 +0200
Subject: Use EmbedLite helper app dialog static registration

Point the helper launcher contract at the EmbedLite ES module so downloads
continue to use the Sailfish save-destination prompt.
---
 toolkit/mozapps/downloads/components.conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/toolkit/mozapps/downloads/components.conf b/toolkit/mozapps/downloads/components.conf
index ccaf5e46c189..64eba64ca262 100644
--- a/toolkit/mozapps/downloads/components.conf
+++ b/toolkit/mozapps/downloads/components.conf
@@ -6,7 +6,7 @@ Classes = [
     {
         'cid': '{F68578EB-6EC2-4169-AE19-8C6243F0ABE1}',
         'contract_ids': ['@mozilla.org/helperapplauncherdialog;1'],
-        'esModule': 'resource://gre/modules/HelperAppDlg.sys.mjs',
-        'constructor': 'nsUnknownContentTypeDialog',
+        'esModule': 'resource://embedlite-components/HelperAppDialog.sys.mjs',
+        'constructor': 'HelperAppLauncherDialog',
     },
 ]
